x-pack/plugins/ml/public/settings/calendar/edit/new_calendar.js

```diff
@@ -0,0 +1,310 @@
+ */
+
+
+
+import React, {
+  Component
+} from 'react';
+import { PropTypes } from 'prop-types';
+
+import {
+  EuiPage,
+  EuiPageContent,
+  EuiOverlayMask,
+} from '@elastic/eui';
+
+import chrome from 'ui/chrome';
+import { getCalendarSettingsData, validateCalendarId } from './utils';
+import { CalendarForm } from './calendar_form';
+import { NewEventModal } from './new_event_modal';
+import { ImportModal } from './import_modal';
+import { ml } from 'plugins/ml/services/ml_api_service';
+import { toastNotifications } from 'ui/notify';
+
+export class NewCalendar extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isNewEventModalVisible: false,
+      isImportModalVisible: false,
+      isNewCalendarIdValid: false,
+      loading: true,
+      jobIds: [],
+      jobIdOptions: [],
+      groupIds: [],
+      groupIdOptions: [],
+      calendars: [],
+      formCalendarId: '',
+      description: '',
+      selectedJobOptions: [],
+      selectedGroupOptions: [],
+      events: [],
+      saving: false,
+      selectedCalendar: undefined,
+    };
+  }
+
+  componentDidMount() {
+    this.formSetup();
+  }
+
+  async formSetup() {
+    try {
+      const { jobIds, groupIds, calendars } = await getCalendarSettingsData();
+
+      const jobIdOptions = jobIds.map((jobId) => ({ label: jobId }));
+      const groupIdOptions = groupIds.map((groupId) => ({ label: groupId }));
+
+      const selectedJobOptions = [];
+      const selectedGroupOptions = [];
+      let eventsList = [];
+      let selectedCalendar;
+      let formCalendarId = '';
+
+      // Editing existing calendar.
+      if (this.props.calendarId !== undefined) {
+        selectedCalendar = calendars.find((cal) => cal.calendar_id === this.props.calendarId);
+
+        if (selectedCalendar) {
+          formCalendarId = selectedCalendar.calendar_id;
+          eventsList = selectedCalendar.events;
+
+          selectedCalendar.job_ids.forEach(id => {
+            if (jobIds.find((jobId) => jobId === id)) {
+              selectedJobOptions.push({ label: id });
+            } else if (groupIds.find((groupId) => groupId === id)) {
+              selectedGroupOptions.push({ label: id });
+            }
+          });
+        }
+      }
+
+      this.setState({
+        events: eventsList,
+        formCalendarId,
+        jobIds,
+        jobIdOptions,
+        groupIds,
+        groupIdOptions,
+        calendars,
+        loading: false,
+        selectedJobOptions,
+        selectedGroupOptions,
+        selectedCalendar
+      });
+    } catch (error) {
+      console.log(error);
+      this.setState({ loading: false });
+      toastNotifications.addDanger('An error occurred loading calendar form data. Try refreshing the page.');
+    }
+  }
+
+  onCreate = async () => {
+    const calendar = this.setUpCalendarForApi();
+    this.setState({ saving: true });
+
+    try {
+      await ml.addCalendar(calendar);
```

Comment:
Is it possible to see if the ID is already in use prior to saving?

Reply:
🤔 We could go through the calendars we've loaded and make sure the id doesn't already exist prior to saving. Is that what you mean?

Reply:
Yes - we do that with the Filters Lists - it gives a clearer message as to why the save failed. Happy for it to go in a separate PR.

Reply:
Adding to follow-up - thanks! 😄